lib/jar.ts

```diff
@@ -97,33 +140,39 @@ async function getMavenPackageInfo(sha1: string): Promise<MavenPackageInfo> {
   });
 }
 
-async function createTempPomFile(
-  filePath: string,
-  pomContents: string,
-): Promise<string> {
+function createTempPomFile(filePath: string, pomContents: string): string {
   try {
     const tmpPom = tmp.fileSync({
       postfix: '-pom.xml',
       dir: path.resolve(path.dirname(filePath)),
     });
-    await fs.writeFileSync(tmpPom.name, pomContents);
+    fs.writeFileSync(tmpPom.name, pomContents);
     return tmpPom.name;
-  } catch (error) {
-    error.message =
-      error.message + '\n\n' + 'Failed to create a temporary pom file.';
-    throw error;
+  } catch (err) {
+    throw new Error('Failed to create temporary pom. ' + err.message);
   }
 }
 
-function getProjectNameFromJarPathParts(root, jarName) {
-  let projectGroupId = path.dirname(jarName);
-  if (path.dirname(jarName) === '.') {
-    // we are in directory of the jar
-    projectGroupId = path.basename(root);
+function getRootDependency(root: string, targetFile?: string): MavenDependency {
+  let groupId;
+  if (targetFile) {
+    groupId = path.dirname(targetFile);
+    if (groupId === '.') {
+      // we are in directory of the jar
+      groupId = path.basename(path.resolve(root));
+    }
+  } else {
+    // take root's parent directory base name
+    groupId = path.basename(path.dirname(path.resolve(root)));
```

Comment:
Note to make sure we properly support `--project-name` option here.

Reply:
will that come through in options?

Reply:
it should happen in the cli